### ISSUE TITLE
2.4.11 Focus Visible (Enhanced) Add user agent exception

### ISSUE THREAD

**Issue by robfentress** [w3c/wcag#1057]
Should there be a qualification added, similar to the one in [1.4.11 Non-text Contrast](https://www.w3.org/WAI/WCAG21/quickref/#non-text-contrast), which says "except ... where the appearance of the component is determined by the user agent and not modified by the author?"

**Comment by JAWS-test** [comment]
No, see https://github.com/w3c/wcag/issues/1003#issuecomment-580689922

**Comment by guyhickling** [comment]
The way the SC is written it requires the developer to create their own focus indicator and not rely on the browser, and I hope that is and will remain the intention. After all it's so ridiculously easy to do, it just requires 3 or 4 lines of CSS! Most designers and developers provide their own anyway these days.

The browsers are never going to change their ways on this one - they have been asked for long enough, and I think it's time to stop waiting on them to do anything. The WCAG should require the developer to provide an indicator until such time (if it ever happens) that the browsers all decide to help low vision users and provide a decent focus indicator as standard. Otherwise, it is people with poor vision who will suffer, and they have been in this position for too many years already.

**Comment by robfentress** [comment]
I certainly understand and am not opposed to that point of view.  I guess I'm just not clear on why an exception is granted for [1.4.11](https://www.w3.org/WAI/WCAG21/quickref/#non-text-contrast), but not [2.4.11](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced.html).  Wouldn't the same general principle apply?

**Comment by guyhickling** [comment]
Rob, In my view the different approach is appropriate simply because of how badly the disabled people concerned are affected. In 1.4.11 Non-text Contrast, firstly, the browsers default indications of fields are mostly of good contrast, either black borders or dark grey. So allowing the browser default in 1.4.11 is not a problem. (Even if the browser default were of poor contrast, there are other things on a form to tell you where the fields are - the text labels. Plus, in most forms, the fields are vertically stacked so once you've found one you know the next one is likely directly under it.)

Contrast that with the focus indicator situation because, in keyboard navigation, the keyboard focus jumps almost randomly all round the page as the user tabs from one link, button, or interactive control to the next. You can never be sure where its going to land next, especially on complex pages with lots of varied content, and especially when some of those items are themselves not so easily visible before they receive focus. For instance some designers seem to delight in making links as unobtrusive as possible, with the minimum indication allowed. Then the browsers come along with their default focus indicators which on some browsers is just a faint dotted line. In other browsers it is a blue line and remains blue even when the page background is blue - so invisibility! 

This is why trying to find the current position of the focus indicator is the single most important problem on the web page if keyboard users can't see it. It's a massive, huge problem on many websites for those keyboard users who, like the rest of us, may well also have poor eyesight, colour blindness, or other vision impairments.

It is this difference in the consequences that makes poor focus indicators different from the effect of poorly contrasting fields on a form (And if you want to know why I feel so strongly about this it's because my own eyesight is degrading at a rate of knots, so I have vested interest, you might say, in encouraging visibility of content as possible! Even though I normally use a mouse, as a website accessibility tester I am constantly having to test using a keyboard and believe me, looking round to see where the focus has gone now is a big problem for me. Just looking for the focus takes most of my time in keyboard testing, if it isn't a high-viz indicator.)

So I hope that explains why allowing browser defaults is ok in one SC but not in this new one. We should not aim for consistency if that causes problems for the very people we are trying to help!

**Comment by alastc** [comment]
Hi @robfentress,

I think @guyhickling has covered it, but just to confirm the intent is to increase the requirement over the browser default. During the discussion on 1.4.11 that was a very hotly contested area, we wanted to include focus styles as part of it. However, the 'adjacent colors' approach suitable for testing icons & graphics is not suitable for testing a _dynamic change_ of contrast. Therefore we needed to create a separate requirement, which is this new criterion.

Does that (with the above comments) answer your question?

**Comment by robfentress** [comment]
Yes, thanks.
